### PR TITLE
Open WorldCat: Update for new site

### LINKED ISSUE #2868: New Open Worldcat format
Reported here: https://forums.zotero.org/discussion/99243/new-worldcat-interface-and-zotero-connector-translator#latest
New interface example e.g. https://worldcat.org/title/1267333854

It doesn't look like they're still offering any RIS or other bib data, but there is quite decent-looking JSON-LD/schema.org in the header, I'd use that? We still don't have an import translator for that, right? So would have to code that into the style?

@AbeJellinek @dstillman any thoughts on the approach here? (would generally be nice to finally support JSON-LD...)

### PR DESCRIPTION
**Pull request by AbeJellinek**
A little rough around the edges, but it grabs pretty much the same metadata that we got before, only with much hackier methods now.

If the translator gets a DOI, it just passes it off to DOI Content Negotiation. The metadata there is much better.

Fixes #2868.

### CODE REVIEW

File: Open WorldCat.js [hunk 1]
```diff
@@ -1,271 +1,202 @@
...
+const RECORD_MAPPING = {
+	oclcNumber: (item, value) => item.extra = (item.extra || '') + `\nOCLC: ${value}`,
+	title: (item, value) => item.title = value.replace(' : ', ': '),
+	edition: 'edition',
+	publisher: 'publisher',
+	publicationPlace: 'place',
+	publicationDate: (item, value) => item.date = ZU.strToISO(value),
+	catalogingLanguage: 'language',
+	summary: 'abstractNote',
+	physicalDescription: 'numPages',
```
[the remark this answers is not in the document]
Comment (AbeJellinek) [reply Open WorldCat.js]: Sure, done.

**Comment by AbeJellinek** [comment]
I'm going to merge this now so that we have something out quickly. @dstillman, let me know if you spot anything wrong and I can work on it further. (One main issue is that detection is much less granular than in the old translator.)

**Comment by dstillman** [comment]
> One main issue is that detection is much less granular than in the old translator.

Meaning item type detection doesn't work as well?

**Comment by dstillman** [comment]
Does this fix the issue with Unicode characters, by any chance?

**Comment by AbeJellinek** [comment]
It just returns `book` for most things rather than the specific type.

**Comment by AbeJellinek** [comment]
But it shouldn't be returning `false` for any valid catalog pages.

**Comment by AbeJellinek** [comment]
> Does this fix the issue with Unicode characters, by any chance?

Yes - or at least I haven't encountered anything that displayed the old issues. [This](https://worldcat.org/title/1014377084) page works fine, for example.

**Comment by dstillman** [comment]
That's great.

**Pull request merged**